pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -282,6 +285,55 @@ func (st *bundleSyncTask) deleteRemovedResources() (retriableError bool, e error
 			continue
 		}
 
+		deletionDelayAnnotation, ok := m.GetAnnotations()[smith.DeletionDelayAnnotation]
+		if ok {
+			// Trigger the "deletion delay" logic
+			deletionDelay, err := time.ParseDuration(deletionDelayAnnotation)
+			if err != nil {
+				return false, errors.Wrap(err, "failed to parse deletion delay duration")
+			}
+			deletionTimestampAnnotation, ok := m.GetAnnotations()[smith.DeletionTimestampAnnotation]
+			if !ok {
+				// Mark object with deletionTimestamp annotation to start the countdown
+				annotations := m.GetAnnotations()
+				deletionTimestampBytes, err := meta_v1.Now().MarshalJSON()
```

Comment:
This is weird. Why not pick a format (take some pre-existing RFC-defined format from stdlib) and do `fmt.Sprintf()` on it? Which format is used should be defined in the spec document.

Reply:
This approach will also spare you from the error below